src/movies.js

```diff
@@ -1,22 +1,55 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const mappedArray = movies.map(function (singleMovie){
+    return singleMovie.director;
+   })
+   return mappedArray;
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies){
+  const movieDirector = movies.filter( (directorName) => {
+    return directorName.director === 'Steven Spielberg';
+  })
+  for (let i=0; movieDirector === 'Steven Spielberg'; i++){
+    const directorDrama = movieDirector.filter( (dramaMovie) => {
+      return dramaMovie.genre === 'Drama';
```

Comment:
Pay attention to the data you are given, movie genre is actually and array, not a string, for example:
genre: ['Crime', 'Drama'],

So you won't be able to compare if dramaMovie.genre === 'Drama'. Instead you need to check if genre array includes 'Drama' string ;)

dramaMovie.genre --> is an array of genres
Drama --> is a single genre

Reply:
that's true...
I would need the position to access the strings or a loop to go through each string